plugins/inputs/openldap/openldap.go

```diff
@@ -0,0 +1,148 @@
+package openldap
+
+import (
+	"crypto/tls"
+	"fmt"
+	"gopkg.in/ldap.v2"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Openldap struct {
+	Host          string
+	Port          int
+	Tls           bool
+	TlsSkipverify bool
+	BindDn        string
+	BindPassword  string
+}
```

Comment:
I didn't implement ldapS because it was never formally standardized, LDAPv3 doesn't specify anything regarding ldaps, only starttls, and go-ldap itself doesn't directly support ldaps.

I'll have to dig into the tls and x509 packages to see about adding a ca cert option.